src/Microsoft.Fx.Portability/Analysis/AnalysisEngine.cs

```diff
@@ -221,6 +224,74 @@ public IEnumerable<string> FindUnreferencedAssemblies(IEnumerable<string> unrefe
             }
         }
 
+        public IList<NuGetPackageInfo> GetNuGetPackagesInfo(IEnumerable<string> assemblies, IEnumerable<FrameworkName> targets)
+        {
+            var nugetPackages = new List<NuGetPackageInfo>();
+            foreach (var assembly in assemblies)
+            {
+                ImmutableDictionary<FrameworkName, IEnumerable<NuGetPackageId>> packages = null;
+                var result = _packageFinder.FindPackage(assembly, targets, out packages);
+                if (result)
+                {
+                    foreach (var target in targets)
+                    {
+                        var nuGetPackageInfo = new NuGetPackageInfo(assembly, target, packages.ContainsKey(target) ? packages[target].ToList() : new List<NuGetPackageId>());
+                        nugetPackages.Add(nuGetPackageInfo);
+                    }
+                }
+            }
+            return nugetPackages;
+        }
+
+        // Get assemblies that should be removed (nuget packages exist for all targets)
+        public List<AssemblyInfo> ComputeAssembliesToRemove(IEnumerable<AssemblyInfo> userAssemblies, IEnumerable<FrameworkName> targets, out List<NuGetPackageInfo> nugetPackagesForUserAssemblies)
+        {
+            var assembliesToRemove = new List<AssemblyInfo>();
+            nugetPackagesForUserAssemblies = new List<NuGetPackageInfo>();
+            foreach (var assembly in userAssemblies)
+            {
+                if (assembly != null)
+                {
+                    ImmutableDictionary<FrameworkName, IEnumerable<NuGetPackageId>> packages = null;
+                    var result = _packageFinder.FindPackage(assembly.AssemblyIdentity, targets, out packages);
+                    if (result)
+                    {
+                        bool supportedOnAllTargets = true;
+                        foreach (var target in targets)
+                        {
+                            var nuGetPackageInfo = new NuGetPackageInfo(assembly.AssemblyIdentity, target, packages.ContainsKey(target) ? packages[target].ToList() : new List<NuGetPackageId>());
```

Comment:
Instead of `packages.ContainsKey(target)` use `packages.TryGet(target, out var r)`.

Reply:
Done